_posts/2025-04-24-Redline-Testing-Comes-to-OpenSearch-Benchmark.md

```diff
@@ -0,0 +1,163 @@
+
+For more technical details, see the [RFC on redline testing](https://github.com/opensearch-project/opensearch-benchmark/issues/785#issue-2898221524).
+
+## Getting started with redline testing
+
+Getting started with redline testing begins with a timed test procedure that defines the duration, target throughput, and number of clients. Once configured, you can run the benchmark with a single command—and optionally customize the maximum number of clients to match your cluster's capacity.
+
+Create a timed test procedure using the following settings:
+
+```json
+{
+  "name": "timed-mode-test-procedure",
+  "schedule": [
+    {
+      "operation": "keyword-terms",
+      "warmup-time-period": {{ warmup_time | default(300) | tojson }},
+      "time-period": {{ time_period | default(900) | tojson }},
+      "target-throughput": {{ target_throughput | default(20) | tojson }},
+      "clients": {{ search_clients | default(20) }}
+    }
+  ]
+}
+```
+
+Then run the benchmark test with the following command:
+
+```bash
+opensearch-benchmark execute-test \
+  --pipeline=benchmark-only \
+  --target-hosts=<your-opensearch-cluster> \
+  --workload=<workload> \
+  --test-procedure=<your-timed-mode-test-procedure> \
+  --redline-test
+```
+
+To override the default max clients (1000), add a number to the `--redline-test` flag, as shown in the following example:
+
+```bash
+--redline-test=1500
+```
+
+Users can customize redline test parameters such as the maximum number of clients, the client ramp-up rate, the percentage of clients to pause during back-off, and the wait time before resuming scale-up.
+
+OpenSearch Benchmark captures the following log information:
+- During the test:
+  - The current number of clients.
+  - The pause/unpause events
+  - The reasons for scaling back
+- After the test:
+  - The maximim number of clients reached
+  - Any performance metrics
+
+
+The following chart shows how redline testing in OpenSearch Benchmark incrementally increases client load during a timed test. Each step represents a controlled ramp-up, allowing the system to observe when performance begins to degrade. In this example image, throughput steadily rises until it plateaus—indicating that the cluster's redline has been reached. This automated feedback loop removes guesswork and enables precise load testing in a single run.
+
+
+![Latency over time](/assets/media/blog-images/2025-04-24-Redline-Testing-Comes-to-OpenSearch-Benchmark/dashboards-latency-over-time.png){: .img-fluid}
+
+## What's next?
+
+Upcoming improvements to redline testing may include:
+
+- Smarter ramp-up strategies, such as binary or exponential search.  
+- Scaling based on latency or service time, not just request failures.  
+
+Redline testing is available now in OpenSearch Benchmark. Try it out and share your feedback by creating a [GitHub issue](https://github.com/opensearch-project/OpenSearch-Benchmark/issues)!
```

Comment:
Redline testing is now available in OpenSearch Benchmark.  We hope you find it useful and look forward to your feedback in the [OpenSearch forum](https://forum.opensearch.org/) or at the OpenSearch Benchmark [community meetup](https://www.meetup.com/opensearch/events/307446531/?eventOrigin=group_upcoming_events).